src/StyledOcticon.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import {COMMON, SystemCommonProps} from './constants'
+import theme from './theme'
+import sx, {SxProp} from './sx'
+import {IconProps} from '@primer/octicons-react'
+import {ComponentProps} from './utils/types'
+
+type OcticonProps = {icon: React.ElementType} & IconProps
+
+function Octicon({icon: IconComponent, ...rest}: OcticonProps) {
+  return <IconComponent {...rest} />
+}
+
+const StyledOcticon = styled(Octicon)<SystemCommonProps & SxProp>`
+  ${COMMON}
+  ${sx}
+`
+
+StyledOcticon.defaultProps = {
+  theme,
+  size: 16
+}
+
+StyledOcticon.propTypes = {
+  ...COMMON.propTypes,
+  ...sx.propTypes,
+  icon: PropTypes.any.isRequired,
+  size: PropTypes.any,
```

Comment:
These propTypes were causing errors. Since we'll be removing propTypes soon, I think `any` is fine for now.

Reply:
I’d personally go ahead and remove PropTypes from this file, but this is fine too!